R/prerun.R

```diff
@@ -1,3 +1,85 @@
+#' Prepare a prerun
+#'
+#' This will take all model files and upload to a specified server and
+#' output the model output zip which can be saved into the ADR or
+#' uploaded into the Naomi app to view plots of model outputs.
+#'
+#' Can use this for countries which cannot get a fit to work via the app
+#' you can prepare a model fit locally and then upload those outputs into
+#' the app.
+#'
+#' @param inputs The model inputs, a named list of file paths including
+#'   pjnz, shape, population, survey and optionally programme and anc.
+#' @param model_output The `hintr_output` object from model fit
+#' @param calibrate_output The `hintr_output` object from calibration
+#' @param server The server URL to upload files to
+#' @param port The port the API is running on
+#' @param output_zip_path The path to save the output zip at, will use
+#'   a tempfile by default
+#'
+#' @return Path to the generated output zip
+#' @export
+hintr_prerun <- function(inputs, model_output, calibrate_output,
+                         server = "https://naomi.unaids.org",
+                         port = "8888",
+                         output_zip_path = tempfile(fileext = ".zip")) {
+  if (!naomi:::is_hintr_output(model_output)) {
+    stop("Model output must be hintr_output object")
+  }
+  if (!naomi:::is_hintr_output(calibrate_output)) {
+    stop("Calibrate output must be hintr_output object")
+  }
+  required <- c("pjnz", "shape", "population", "survey")
+  optional <- c("programme", "anc")
+  assert_names(inputs, required, optional)
+  assert_files_exist(inputs)
+
+  if (!is.null(port)) {
+    url <- paste0(server, ":", port)
+  } else {
+    url <- server
+  }
+
+  uploaded_inputs <- lapply(inputs, function(input) {
+    filename <- basename(input)
+    res <- httr::POST(paste0(url, "/internal/upload/input/", filename),
```

Comment:
do we have a way of querying to see which of these are needed to be uploaded? or I guess if this is only ever on the imperial network that's not really a big problem...

Reply:
Not at the moment, there is a ticket for that https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-3887/Add-endpoint-to-check-if-files-exist would be good to do.